ddtrace/tracer/option.go

```diff
@@ -293,6 +293,7 @@ func newConfig(opts ...StartOption) *config {
 		c.hostname, err = os.Hostname()
 		if err != nil {
 			log.Warn("unable to look up hostname: %v", err)
+			return c, err
```

Comment:
Similarly here, I would wrap the error with similar text as in the log.Warn above. Otherwise, tracer.Start might return something like `sysctl kern.hostname :  err` which is not very useful for customers